docs/segments.md

```diff
@@ -0,0 +1,27 @@
+---
+category: DevelopInDepth
+title: Segments
+---
+# Matomo core - Segments
+
+## How are segments encoded?
+
+Segments are typically encoded three times:
+
+* The first decode is on the whole segment: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Segment.php#L104 . The corresponding encode was added in 3.6.0 here: https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/Live/javascripts/SegmentedVisitorLog.js#L123
+* the second decode is on individual conditions (ie, pageUrl==...): https://github.com/matomo-org/matomo/blob/3.x-dev/core/Segment/SegmentExpression.php#L91
+* the third decode is on individual values: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Segment/SegmentExpression.php#L112
```

Comment:
Are the links on 3.x-dev branch on purpose?

Reply:
@sgiehl it was copied from a PR description. I guess it's fine to use 3.x because the code basically doesn't change anymore there. Could otherwise reference a 4.X version but in the end there's no difference